keps/sig-release/20190316-rebase-images-to-distroless.md

```diff
@@ -0,0 +1,149 @@
+#### Type 2 Debian Based Images
+An image can be based from Debian due to different reasons.
+
+- One big reason is that the image needs to use shell to redirect the glog. This base image now is an overkill because K8s 1.13 can support using klog which accepts a --log-file flag to point to the log path directly. (Historically images doing this mostly relied on busybox or alpine. Some recent change has migrated off those to debian-base. [PR/70245](https://github.com/kubernetes/kubernetes/pull/70245/files))
+- Another reason of using debian is from the CVE concerns. Those images are originally rebased from busybox to debian for better CVE feeds and management (See [PR/70245](https://github.com/kubernetes/kubernetes/pull/70245)).
+- A third type of images uses debian for certain external dependencies. For example, kube-proxy has a hard requirement for dynamic-link which uses glibc.
+#### Type 3 Alpine Based Images
+The reasons for images based on alpine are similar to the ones on debian. Debian is more widely used due to previous “Establish base image policy for system containers” effort (see [issue/40248](https://github.com/kubernetes/kubernetes/issues/40248)).
+### Distroless and Previous Work
+"Distroless images contain only your application and its runtime dependencies. They do not contain package managers, shells or any other programs you would expect to find in a standard Linux distribution.”  (from [github.com/GoogleContainerTools/distroless/README](https://github.com/GoogleContainerTools/distroless))
+Distroless supports the dependencies where “FROM scratch” misses and more light-weighted than debian or alpine. Meanwhile, distroless “improves the signal to noise of scanners (e.g. CVE) and reduces the burden of establishing provenance to just what you need.”(from [github.com/GoogleContainerTools/distroless/README](https://github.com/GoogleContainerTools/distroless))
+
+Using Distroless as a common image base is originally proposed as an exploration area in [the base image policy for system containers](https://github.com/kubernetes/kubernetes/issues/40248). Tim(tallclair@) has driven the effort on defining and establishing the base image policy (main changes):
+
+* Add Alpine iptable as base image for kube-proxy. Previously kube-proxy is based on debian iptable image.  (This direction is scrapped, see [issue/39696](https://github.com/kubernetes/kubernetes/issues/39696) for details)
+Rebase busybox images to debian-base.
+* Rebase busybox images to debian-base.
+* Rebase certain alpine images to debian-base
+* Image related works on migrating apiserver-builder to kube-builder
```

Comment:
what does this mean? I don't think I ever proposed this :)

Reply:
I looked at all the comments in the issue 39696, and this [PR ](https://github.com/kubernetes-sigs/cluster-api/pull/494) is labeled as referencing the issue.

Removed this item since it's irrelevant.